### DISCUSSION TITLE
Training on Multiple Datasets

### DISCUSSION THREAD

**Discussion by ZikangZhou** [pyg-team/pytorch_geometric#3275] [Q&A]
I have two datasets, lets' say dataset A and dataset B, where each sample is a small graph. Dataset A contains more samples than dataset B, let's say the ratio is 3:1. 
I want to train a GNN model using these two datasets in a mini-batch manner. I have implemented torch_geometric.data.Dataset for both A and B, but how can I concatenate them so that each batch samples 3 / 4 graphs from A and 1 / 4 graphs from B?

**Comment by rusty1s** [comment]
You can use `torch.utils.data.ConcatDataset` to concatenate datasets. For over-sampling examples in dataset B, you can concatenate dataset B four times. For under-sampling, you need to implement a customized PyTorch sampler, see https://discuss.pytorch.org/t/balanced-sampling-between-classes-with-torchvision-dataloader/2703/3.
